### PR TITLE
Ensure that callbacks fire in a FIFO order

### PR DESCRIPTION
**Pull request by jcheng5**
Version bump required so Leaflet can detect this fix

cc @wch

### CODE REVIEW

File: R/utils.R [hunk 1]
```diff
@@ -580,7 +580,11 @@ Callbacks <- R6Class(
       })
     },
     invoke = function(..., onError=NULL) {
-      for (callback in .callbacks$values()) {
+      # Ensure that calls are invoked in the order that they were registered
+      keys <- as.character(sort(as.integer(.callbacks$keys()), decreasing = TRUE))
```
Comment (wch) [inline R/utils.R]: Yuck... but I don't see a better way to do this.

Comment (jcheng5) [inline R/utils.R]: Yep. FWIW, ordered lists with fast random-access removal always seem to be pretty gross.

**Comment by wch** [comment]
Would it make sense to use a queue instead of a map to store the callbacks? I have a pretty good implementation here (see Queue2 in the README and R code):
  https://github.com/wch/qstack/

**Comment by wch** [comment]
I should mention that the Stack in shiny is basically Stack2 from that package, adapted to be an R6 class.

**Comment by jcheng5** [comment]
No, because random-access deletes are important.

**Comment by wch** [comment]
OK, looks good then.

**Pull request merged**